### PR TITLE
2939 multiline commit message

### LINKED ISSUE #2939: Multiple-Line commit message
- Gitea version (or commit ref): 1.3.0+rc1
- Git version: 2.7.4
- Operating system: Ubuntu 16.04
- Database (use `[x]`):
  - [x] MySQL
- Can you reproduce the bug at https://try.gitea.io:
  - [x] No: Didn't find a matching commit

## Description
Only the first line of a commit message is shown.

## Screenshots
How it looks in gitea:
Commit history -> https://i.imgur.com/oVDhmAP.png
Commit view -> https://i.imgur.com/lXAfOxV.png

The whole commit message (from a discord webhook):
https://i.imgur.com/9BeVaIS.png

### PR DESCRIPTION
**Pull request by makarchuk**
View of multiline commit messages as tooltips. 
Fixes #2939

### CODE REVIEW

File: templates/repo/commits_table.tmpl [hunk 1]
```diff
@@ -60,7 +60,7 @@
 							</a>
 						</td>
 						<td class="message collapsing">
-							<span class="has-emoji{{if gt .ParentCount 1}} grey text{{end}}">{{RenderCommitMessage .Summary $.RepoLink $.Repository.ComposeMetas}}</span>
+							<span class="has-emoji{{if gt .ParentCount 1}} grey text{{end}}" title={{.Message}}>{{RenderCommitMessage .Summary $.RepoLink $.Repository.ComposeMetas}}</span>
```
Comment (lafriks) [inline templates/repo/commits_table.tmpl]: Quotes seems to be missing

**Review by lafriks — CHANGES_REQUESTED** [review]
Also title should be added `page.tmpl` element `<h3>` where commit message is rendered

File: templates/repo/commits_table.tmpl (hunk 1 above)
Comment (makarchuk) [reply templates/repo/commits_table.tmpl]: I added the quotes and title in `page.tmpl`

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
It still seems strange to me that attribute value contains newlines. It needs to be verified that it works on all supported browsers

**Comment by makarchuk** [comment]
If it helps: These are my screenshots from chrome and Firefox (newest versions both) on my ubuntu
screenshot from 2017-11-22 00-35-51
screenshot from 2017-11-22 00-33-34
screenshot from 2017-11-22 00-35-30
screenshot from 2017-11-22 00-35-17

**Comment by lafriks** [comment]
Yes on these were also I tested :) I'm more worried about Safari and IE on windows. These are usually the most problematic ones

**Comment by sondr3** [comment]
I don't really like the idea of viewing multiline commits in a tooltip, both GitHub and GitLab show multiline commits with a `...` button you press to expand the commit message into full view. First off it makes viewing the history harder if you want to open multiple commits and view their messages, you can't easily copy commit messages or potential names or emails from contributor/whoever signed off on the commit and the discoverability is poor, I would never consider hovering over commit messages to get the full body in a tooltip, if it was a multiline commit I'd expect there to be some kind of sign that enables me to view the full message.

**Comment by makarchuk** [comment]
I can try and implement this, but I'm new to a project, so I might need some guidance.  
I haven't found any information on how we cook JS. Is it just good old JQuery(So I can just modify in project files and it'll be all good) or some kind of fancy modern framework?   
Is it acceptable to use `onclick` on HTML elements or I should use jq `$('SELECTORHERE').click(...)`?

**Comment by lunny** [comment]
@makarchuk you can use vue.js

**Comment by sondr3** [comment]
I'd love to help but I'm in the middle of all my exams for this semester, I can help over Christmas if this is still open 😃

**Comment by lafriks** [comment]
@sondr3 jQuery for this would be ok

**Comment by sondr3** [comment]
I've created my own version of multiline commits in #2980, but the implementation is very rough and not finished, I'd love if any of you would give me a helping hand.

**Comment by sondr3** [comment]
My own version is now done (or so I hope), if any of you would like to test it I'd really appreciate it.

**Comment by lafriks** [comment]
Closing this as other PR has implemented this with expandable button

**Pull request closed without merging**